packages/ragbits-document-search/examples/documents_chat.py

```diff
@@ -0,0 +1,171 @@
+    DATABASE_CREATION_STATUS_PLACEHOLDER = "Upload files and click 'Create Database' to start..."
+    DATABASE_LOADING_STATUS_LABEL = "Database loading status"
+    DATABASE_LOADING_STATUS_PLACEHOLDER = "Click 'Load Database' to start..."
+
+    def __init__(
+        self,
+        database_path: str = "chroma",
+        index_name: str = "documents",
+        model_name: str = "gpt-4o-2024-08-06",
+        columns_ratios: tuple[int, int] = (1, 5),
+        chatbot_height_vh: int = 90,
+    ) -> None:
+        self._database_path = database_path
+        self._index_name = index_name
+        self._columns_ratios = columns_ratios
+        self._chatbot_height_vh = chatbot_height_vh
+        self._documents_ingested = False
+        self._prepare_document_search(self._database_path, self._index_name)
+        self._llm = LiteLLM(model_name, use_structured_output=True)
+
+    def _prepare_document_search(self, database_path: str, index_name: str) -> None:
+        chroma_client = chromadb.PersistentClient(path=database_path)
+        embedding_client = LiteLLMEmbeddings()
+
+        vector_store = ChromaDBStore(
+            index_name=index_name,
+            chroma_client=chroma_client,
+            embedding_function=embedding_client,
+        )
+        self.document_search = DocumentSearch(embedder=vector_store.embedding_function, vector_store=vector_store)
+
+    async def _create_database(self, document_paths: list[str]) -> str:
+        for path in document_paths:
+            await self.document_search.ingest_document(DocumentMeta.from_local_path(Path(path)))
+        self._documents_ingested = True
+        return self.DATABASE_CREATED_MESSAGE + self._database_path
+
+    def _load_database(self, database_path: str) -> str:
+        self._prepare_document_search(database_path, self._index_name)
+        self._documents_ingested = True
+        return self.DATABASE_LOADED_MESSAGE
+
+    async def _handle_message(
+        self, message: str, history: list[list[str]]  # pylint: disable=unused-argument
+    ) -> AsyncIterator[str]:
+        if not self._documents_ingested:
+            yield self.NO_DOCUMENTS_INGESTED_MESSAGE
+        results = await self.document_search.search(message)
+        prompt = RAGPrompt(
+            QueryWithContext(query=message, context=[i.content for i in results if isinstance(i, TextElement)])
+        )
+        response = await self._llm.generate(prompt)
+        yield response.answer
+
+    def prepare_layout(self) -> gr.Blocks:
+        """
+        Crates gradio layout as gr.Blocks and connects component events with proper handlers
+
+        Returns:
+            gradio layout
+        """
+        with gr.Blocks(fill_height=True, fill_width=True) as app:
+            with gr.Row():
+                with gr.Column(scale=self._columns_ratios[0]):
+                    with gr.Group():
+                        documents_picker = gr.File(file_count="directory", label=self.DOCUMENT_PICKER_LABEL)
```

Comment:
I think it should be `"multiple"` - right now it forces user to upload full directory